packages/runtime/container-runtime/src/scheduleManager.ts

```diff
@@ -18,6 +22,36 @@ type IRuntimeMessageMetadata = undefined | {
     batch?: boolean;
 };
 
+/**
+ * The internal state of the batch schedule
+ *
+ * Individual - processing an individual message
+ * Batch - processing a message which is part of a batch
+ * Error - encountered an error between processing ops
+ * None - In between batches or individual messages
+ */
+type BatchProcessingState = "Individual" | "Batch" | "Error" | "None";
+
+/**
+ * Relative to a batch, a message can be:
+ *
+ * StartOfBatch - the first message in a batch, marking the start
+ * EndOfBatch - the last message in a batch, marking the end
+ * Individual - part of a batch (including a one message batch)
+ */
+type MessageBatchState = "StartOfBatch" | "EndOfBatch" | "Individual";
```

Comment:
I have not (yet) looked through the rest of the PR, but I'd love us to move toward direction where PendingStateManager operates only with one entity - batch of ops. I.e. there are no start or end of batch records, there are no flushes or changes in flush mode - the only language that is used between ContainerRuntime - PendingStateManager is "batch of ops".

See feedback in https://github.com/microsoft/FluidFramework/pull/11659 and https://dev.azure.com/fluidframework/internal/_queries/edit/1833